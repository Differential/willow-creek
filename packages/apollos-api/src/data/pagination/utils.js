// eslint-disable-next-line import/prefer-default-export
export const withEdgePagination = async ({ edges }) => {
  const result = await edges;
  const startCursor = result.length ? result[0].cursor : null;
  const endCursor = result.length ? result[result.length - 1].cursor : null;
  return { startCursor, endCursor };
};
