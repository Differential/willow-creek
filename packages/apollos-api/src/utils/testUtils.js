import { KeyValueCache } from 'apollo-server-caching';
import {
  context as getContext,
  dataSources as getDataSources,
} from '../data/index';

export function getTestContext(req) {
  const context = getContext(req);
  const dataSources = getDataSources();
  // Apollo Server does this internally.
  Object.values(dataSources).forEach((dataSource) => {
    if (dataSource.initialize) {
      dataSource.initialize({ context, cache: KeyValueCache });
    }
  });
  context.dataSources = dataSources;
  return context;
}

export const buildGetMock = (response, dataSource) => {
  const get = jest.fn();
  if (Array.isArray(response) && Array.isArray(response[0])) {
    response.forEach((responseVal) => {
      get.mockReturnValueOnce(
        new Promise((resolve) => resolve(dataSource.normalize(responseVal)))
      );
    });
  }
  get.mockReturnValue(
    new Promise((resolve) => resolve(dataSource.normalize(response)))
  );
  return get;
};
