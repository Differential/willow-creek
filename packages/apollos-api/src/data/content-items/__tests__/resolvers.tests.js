import { graphql } from 'graphql';
import { fetch } from 'apollo-server-env';
import { makeExecutableSchema } from 'apollo-server';

import { createGlobalId } from '@apollosproject/server-core';
import ApollosConfig from '@apollosproject/config';
import { testSchema as typeDefs, resolvers } from '../..';
import { resolver } from '..';
import { getTestContext } from '../../../utils/testUtils';
// we import the root-level schema and resolver so we test the entire integration:

ApollosConfig.loadJs({
  ROCK: {
    API_URL: 'https://apollosrock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
  },
  ROCK_CONSTANTS: {
    IMAGE: 10,
    AUDIO_FILE: 77,
    VIDEO_FILE: 79,
  },
  ROCK_MAPPINGS: {
    FEED_CONTENT_CHANNEL_IDS: [1, 2, 3, 4, 6, 8],
  },
});

const contentItemFragment = `
  fragment ContentItemFragment on UniversalContentItem {
    id
    __typename
    title
    coverImage {
      name
      key
      sources {
        uri
      }
    }
    images {
      __typename # Typenames here to increase test coverage
      name
      key
      sources {
        __typename
        uri
      }
    }
    videos {
      __typename
      name
      key
      sources {
        __typename
        uri
      }
      embedHtml
    }
    audios {
      __typename
      name
      key
      sources {
        __typename
        uri
      }
    }
    htmlContent
    childContentItemsConnection {
      edges {
        node {
          id
          __typename
        }
        cursor
      }
      pageInfo {
        startCursor
        endCursor
      }
    }
    parentChannel {
      id
      __typename
    }
    terms {
      key
      value
    }
    sharing {
      __typename
      url
      title
      message
    }
  }
`;

describe('UniversalContentItem', () => {
  let schema;
  let context;
  beforeEach(() => {
    fetch.resetMocks();
    fetch.mockRockDataSourceAPI();
    schema = makeExecutableSchema({ typeDefs, resolvers });
    context = getTestContext();
  });

  it('gets a user feed', async () => {
    const query = `
      query {
        userFeed {
          edges {
            node {
              ...ContentItemFragment
            }
          }
        }
      }
      ${contentItemFragment}
    `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('gets a content item', async () => {
    const query = `
      query {
        node(id: "${createGlobalId(1, 'UniversalContentItem')}") {
          ...ContentItemFragment
        }
      }
      ${contentItemFragment}
    `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('gets a devotional item', async () => {
    const query = `
      query {
        node(id: "${createGlobalId(123, 'DevotionalContentItem')}") {
          id
          ... on DevotionalContentItem {
            id
            title
            scriptures {
              html
            }
          }
        }
      }
    `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it("gets a content item and it's siblings", async () => {
    const query = `
      query {
        userFeed {
          edges {
            node {
              ...ContentItemFragment
              ... on UniversalContentItem {
                siblingContentItemsConnection {
                        edges {
                    node {
                      id
                      __typename
                    }
                    cursor
                  }
                }
              }
            }
          }
        }
      }
      ${contentItemFragment}
    `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('properly handles empty attribute values', async () => {
    const query = `
      query {
        node(id: "${createGlobalId(
          'test-case-no-attributes',
          'UniversalContentItem'
        )}") {
          ...ContentItemFragment
        }
      }
      ${contentItemFragment}
    `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('filters terms by a match string', async () => {
    const query = `
    query {
      node(id: "${createGlobalId(1, 'UniversalContentItem')}") {
        ...on UniversalContentItem {
          terms(match: "speaker") {
            value
          }
        }
      }
    }
  `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });
});

const { ContentItemsConnection } = resolver;

describe('ContentItemsConnection resolvee', () => {
  it('builds a pageInfo object with items', async () => {
    const edges = Promise.resolve([
      { cursor: `item-0` },
      { cursor: `item-1` },
      { cursor: `item-2` },
    ]);
    const { startCursor, endCursor } = await ContentItemsConnection.pageInfo({
      edges,
    });

    expect(startCursor).toEqual('item-0');
    expect(endCursor).toEqual('item-2');
  });
  it('builds a pageInfo object without items', async () => {
    const edges = [];
    const { startCursor, endCursor } = await ContentItemsConnection.pageInfo({
      edges,
    });

    expect(startCursor).toEqual(null);
    expect(endCursor).toEqual(null);
  });
});
