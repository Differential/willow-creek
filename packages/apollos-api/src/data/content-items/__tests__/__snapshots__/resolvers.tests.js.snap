// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniversalContentItem filters terms by a match string 1`] = `
Object {
  "data": Object {
    "node": Object {
      "terms": Array [
        Object {
          "value": "Pete Foster",
        },
      ],
    },
  },
}
`;

exports[`UniversalContentItem gets a content item 1`] = `
Object {
  "data": Object {
    "node": Object {
      "__typename": "UniversalContentItem",
      "audios": Array [
        Object {
          "__typename": "AudioMedia",
          "key": "audioLink",
          "name": "Audio Link",
          "sources": Array [
            Object {
              "__typename": "AudioMediaSource",
              "uri": "https://rockrms.blob.core.windows.net/sampledata/podcasting/sample.mp3",
            },
          ],
        },
      ],
      "childContentItemsConnection": Object {
        "edges": Array [
          Object {
            "cursor": "b487224762b030f470967f45d7205823",
            "node": Object {
              "__typename": "UniversalContentItem",
              "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
            },
          },
        ],
        "pageInfo": Object {
          "endCursor": "b487224762b030f470967f45d7205823",
          "startCursor": "b487224762b030f470967f45d7205823",
        },
      },
      "coverImage": Object {
        "key": "image",
        "name": "Image",
        "sources": Array [
          Object {
            "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=0241ED2F-B527-424C-917C-1142A398711F",
          },
        ],
      },
      "htmlContent": "<p>
	Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sollicitudin condimentum aliquet. In est nulla, lacinia ac dictum et, laoreet vitae elit. Proin tempus tellus ligula, a consequat diam consectetur a. Phasellus luctus velit sed lorem mollis commodo. Nunc sit amet blandit velit. Donec tincidunt congue facilisis. Sed iaculis at neque non porttitor. Phasellus ultrices egestas erat feugiat pellentesque. Duis venenatis, dolor quis fringilla tempus, sem lorem euismod lectus, sed egestas felis magna at felis. Pellentesque ut rhoncus erat, a pulvinar purus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Ut sit amet consequat est. Maecenas et porta dui, non condimentum lectus.</p>
<p>
	Suspendisse vel nibh odio. Pellentesque porta sapien ligula, in laoreet diam tempus sed. Morbi nunc erat, mattis eu pulvinar blandit, adipiscing quis magna. Ut quis dui lobortis velit suscipit consectetur. Nulla iaculis fermentum egestas. Aenean venenatis sagittis mauris, sed rhoncus purus accumsan ac. Suspendisse potenti. Sed sed tempor turpis. Duis sit amet nisi nec purus fringilla condimentum. Phasellus non lacus arcu. Donec scelerisque, erat sed tempor elementum, nulla risus scelerisque ante, ac imperdiet velit magna ut quam. Nam tristique orci auctor consequat laoreet. Quisque malesuada metus sed sodales eleifend. Aenean rhoncus, mi sit amet ullamcorper tincidunt, sem sem rutrum felis, in semper enim massa ut sem.</p>
<p>
	Vivamus diam urna, cursus in sapien in, porta gravida enim. Cras non fringilla arcu, tincidunt laoreet lacus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Aliquam volutpat felis quis augue faucibus ultrices. Morbi lobortis vestibulum sodales. Sed tincidunt urna vitae felis ultrices, pharetra placerat quam dignissim. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vel adipiscing tellus. In vitae sodales diam. Aliquam pharetra orci a porta molestie. In et neque bibendum, viverra leo sit amet, auctor magna. Morbi posuere massa sed metus euismod, et adipiscing sem dictum. Cras eget elementum risus, non imperdiet ligula.</p>
",
      "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
      "images": Array [
        Object {
          "__typename": "ImageMedia",
          "key": "image",
          "name": "Image",
          "sources": Array [
            Object {
              "__typename": "ImageMediaSource",
              "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=0241ED2F-B527-424C-917C-1142A398711F",
            },
          ],
        },
        Object {
          "__typename": "ImageMedia",
          "key": "detailImage",
          "name": "Detail Image",
          "sources": Array [
            Object {
              "__typename": "ImageMediaSource",
              "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=3DA90982-118A-4BFE-9A32-58D9F610090A",
            },
          ],
        },
        Object {
          "__typename": "ImageMedia",
          "key": "imageByUrlTest",
          "name": "Image By Url Test",
          "sources": Array [
            Object {
              "__typename": "ImageMediaSource",
              "uri": "https://some-domain.com/some/path/to/image.jpg",
            },
          ],
        },
        Object {
          "__typename": "ImageMedia",
          "key": "imageBySchemalessUrlTest",
          "name": "Image By Schemaless Url Test",
          "sources": Array [
            Object {
              "__typename": "ImageMediaSource",
              "uri": "https://some-domain.com/some/path/to/image.jpg",
            },
          ],
        },
        Object {
          "__typename": "ImageMedia",
          "key": "imageUnknownFormat",
          "name": "Image Of Unknown format",
          "sources": Array [
            Object {
              "__typename": "ImageMediaSource",
              "uri": "ftp://some-domain.com/some/path/to/image.jpg",
            },
          ],
        },
      ],
      "parentChannel": Object {
        "__typename": "ContentChannel",
        "id": "ContentChannel:744e1011cfa6a90bb2fbfeb3ecdc1451",
      },
      "sharing": Object {
        "__typename": "SharableContentItem",
        "message": "",
        "title": "SAMPLE: Easter",
        "url": "https://apollosrock.newspring.cc/",
      },
      "terms": Array [
        Object {
          "key": "summaryText",
          "value": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sollicitudin condimentum aliquet. In est nulla, lacinia ac dictum et, laoreet vitae elit. Proin tempus tellus ligula, a consequat diam consectetur a.",
        },
        Object {
          "key": "campuses",
          "value": "",
        },
        Object {
          "key": "primaryAudience",
          "value": "B364CDEE-F000-4965-AE67-0C80DDA365DC",
        },
        Object {
          "key": "secondaryAudiences",
          "value": "",
        },
        Object {
          "key": "speaker",
          "value": "Pete Foster",
        },
      ],
      "title": "SAMPLE: Easter",
      "videos": Array [
        Object {
          "__typename": "VideoMedia",
          "embedHtml": "<script src=\\"//fast.wistia.com/embed/medias/kqykx8xbzq.jsonp\\" async></script><script src=\\"//fast.wistia.com/assets/external/E-v1.js\\" async></script><div class=\\"wistia_responsive_padding\\" style=\\"padding:56.25% 0 0 0;position:relative;\\"><div class=\\"wistia_responsive_wrapper\\" style=\\"height:100%;left:0;position:absolute;top:0;width:100%;\\"><div class=\\"wistia_embed wistia_async_kqykx8xbzq videoFoam=true\\" style=\\"height:100%;width:100%\\">&nbsp;</div></div></div>",
          "key": "videoLink",
          "name": "Video Link",
          "sources": Array [
            Object {
              "__typename": "VideoMediaSource",
              "uri": "https://rockrms.blob.core.windows.net/sampledata/podcasting/money-wise.mp4",
            },
          ],
        },
      ],
    },
  },
}
`;

exports[`UniversalContentItem gets a content item and it's siblings 1`] = `
Object {
  "data": Object {
    "userFeed": Object {
      "edges": Array [
        Object {
          "node": Object {
            "__typename": "UniversalContentItem",
            "audios": Array [
              Object {
                "__typename": "AudioMedia",
                "key": "audioLink",
                "name": "Audio Link",
                "sources": Array [
                  Object {
                    "__typename": "AudioMediaSource",
                    "uri": "https://rockrms.blob.core.windows.net/sampledata/podcasting/sample.mp3",
                  },
                ],
              },
            ],
            "childContentItemsConnection": Object {
              "edges": Array [
                Object {
                  "cursor": "b487224762b030f470967f45d7205823",
                  "node": Object {
                    "__typename": "UniversalContentItem",
                    "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
                  },
                },
              ],
              "pageInfo": Object {
                "endCursor": "b487224762b030f470967f45d7205823",
                "startCursor": "b487224762b030f470967f45d7205823",
              },
            },
            "coverImage": Object {
              "key": "image",
              "name": "Image",
              "sources": Array [
                Object {
                  "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=0241ED2F-B527-424C-917C-1142A398711F",
                },
              ],
            },
            "htmlContent": "<p>
	Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sollicitudin condimentum aliquet. In est nulla, lacinia ac dictum et, laoreet vitae elit. Proin tempus tellus ligula, a consequat diam consectetur a. Phasellus luctus velit sed lorem mollis commodo. Nunc sit amet blandit velit. Donec tincidunt congue facilisis. Sed iaculis at neque non porttitor. Phasellus ultrices egestas erat feugiat pellentesque. Duis venenatis, dolor quis fringilla tempus, sem lorem euismod lectus, sed egestas felis magna at felis. Pellentesque ut rhoncus erat, a pulvinar purus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Ut sit amet consequat est. Maecenas et porta dui, non condimentum lectus.</p>
<p>
	Suspendisse vel nibh odio. Pellentesque porta sapien ligula, in laoreet diam tempus sed. Morbi nunc erat, mattis eu pulvinar blandit, adipiscing quis magna. Ut quis dui lobortis velit suscipit consectetur. Nulla iaculis fermentum egestas. Aenean venenatis sagittis mauris, sed rhoncus purus accumsan ac. Suspendisse potenti. Sed sed tempor turpis. Duis sit amet nisi nec purus fringilla condimentum. Phasellus non lacus arcu. Donec scelerisque, erat sed tempor elementum, nulla risus scelerisque ante, ac imperdiet velit magna ut quam. Nam tristique orci auctor consequat laoreet. Quisque malesuada metus sed sodales eleifend. Aenean rhoncus, mi sit amet ullamcorper tincidunt, sem sem rutrum felis, in semper enim massa ut sem.</p>
<p>
	Vivamus diam urna, cursus in sapien in, porta gravida enim. Cras non fringilla arcu, tincidunt laoreet lacus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Aliquam volutpat felis quis augue faucibus ultrices. Morbi lobortis vestibulum sodales. Sed tincidunt urna vitae felis ultrices, pharetra placerat quam dignissim. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vel adipiscing tellus. In vitae sodales diam. Aliquam pharetra orci a porta molestie. In et neque bibendum, viverra leo sit amet, auctor magna. Morbi posuere massa sed metus euismod, et adipiscing sem dictum. Cras eget elementum risus, non imperdiet ligula.</p>
",
            "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
            "images": Array [
              Object {
                "__typename": "ImageMedia",
                "key": "image",
                "name": "Image",
                "sources": Array [
                  Object {
                    "__typename": "ImageMediaSource",
                    "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=0241ED2F-B527-424C-917C-1142A398711F",
                  },
                ],
              },
              Object {
                "__typename": "ImageMedia",
                "key": "detailImage",
                "name": "Detail Image",
                "sources": Array [
                  Object {
                    "__typename": "ImageMediaSource",
                    "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=3DA90982-118A-4BFE-9A32-58D9F610090A",
                  },
                ],
              },
              Object {
                "__typename": "ImageMedia",
                "key": "imageByUrlTest",
                "name": "Image By Url Test",
                "sources": Array [
                  Object {
                    "__typename": "ImageMediaSource",
                    "uri": "https://some-domain.com/some/path/to/image.jpg",
                  },
                ],
              },
              Object {
                "__typename": "ImageMedia",
                "key": "imageBySchemalessUrlTest",
                "name": "Image By Schemaless Url Test",
                "sources": Array [
                  Object {
                    "__typename": "ImageMediaSource",
                    "uri": "https://some-domain.com/some/path/to/image.jpg",
                  },
                ],
              },
              Object {
                "__typename": "ImageMedia",
                "key": "imageUnknownFormat",
                "name": "Image Of Unknown format",
                "sources": Array [
                  Object {
                    "__typename": "ImageMediaSource",
                    "uri": "ftp://some-domain.com/some/path/to/image.jpg",
                  },
                ],
              },
            ],
            "parentChannel": Object {
              "__typename": "ContentChannel",
              "id": "ContentChannel:744e1011cfa6a90bb2fbfeb3ecdc1451",
            },
            "sharing": Object {
              "__typename": "SharableContentItem",
              "message": "",
              "title": "SAMPLE: Easter",
              "url": "https://apollosrock.newspring.cc/",
            },
            "siblingContentItemsConnection": Object {
              "edges": Array [
                Object {
                  "cursor": "b487224762b030f470967f45d7205823",
                  "node": Object {
                    "__typename": "UniversalContentItem",
                    "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
                  },
                },
              ],
            },
            "terms": Array [
              Object {
                "key": "summaryText",
                "value": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sollicitudin condimentum aliquet. In est nulla, lacinia ac dictum et, laoreet vitae elit. Proin tempus tellus ligula, a consequat diam consectetur a.",
              },
              Object {
                "key": "campuses",
                "value": "",
              },
              Object {
                "key": "primaryAudience",
                "value": "B364CDEE-F000-4965-AE67-0C80DDA365DC",
              },
              Object {
                "key": "secondaryAudiences",
                "value": "",
              },
              Object {
                "key": "speaker",
                "value": "Pete Foster",
              },
            ],
            "title": "SAMPLE: Easter",
            "videos": Array [
              Object {
                "__typename": "VideoMedia",
                "embedHtml": "<script src=\\"//fast.wistia.com/embed/medias/kqykx8xbzq.jsonp\\" async></script><script src=\\"//fast.wistia.com/assets/external/E-v1.js\\" async></script><div class=\\"wistia_responsive_padding\\" style=\\"padding:56.25% 0 0 0;position:relative;\\"><div class=\\"wistia_responsive_wrapper\\" style=\\"height:100%;left:0;position:absolute;top:0;width:100%;\\"><div class=\\"wistia_embed wistia_async_kqykx8xbzq videoFoam=true\\" style=\\"height:100%;width:100%\\">&nbsp;</div></div></div>",
                "key": "videoLink",
                "name": "Video Link",
                "sources": Array [
                  Object {
                    "__typename": "VideoMediaSource",
                    "uri": "https://rockrms.blob.core.windows.net/sampledata/podcasting/money-wise.mp4",
                  },
                ],
              },
            ],
          },
        },
      ],
    },
  },
}
`;

exports[`UniversalContentItem gets a devotional item 1`] = `
Object {
  "data": Object {
    "node": Object {
      "id": "DevotionalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
      "scriptures": Array [
        Object {
          "html": "<p class=\\"p\\"><span data-number=\\"1\\" class=\\"v\\">1</span>The Song of songs, which is Solomon’s.</p>",
        },
      ],
      "title": "SAMPLE: Easter",
    },
  },
}
`;

exports[`UniversalContentItem gets a user feed 1`] = `
Object {
  "data": Object {
    "userFeed": Object {
      "edges": Array [
        Object {
          "node": Object {
            "__typename": "UniversalContentItem",
            "audios": Array [
              Object {
                "__typename": "AudioMedia",
                "key": "audioLink",
                "name": "Audio Link",
                "sources": Array [
                  Object {
                    "__typename": "AudioMediaSource",
                    "uri": "https://rockrms.blob.core.windows.net/sampledata/podcasting/sample.mp3",
                  },
                ],
              },
            ],
            "childContentItemsConnection": Object {
              "edges": Array [
                Object {
                  "cursor": "b487224762b030f470967f45d7205823",
                  "node": Object {
                    "__typename": "UniversalContentItem",
                    "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
                  },
                },
              ],
              "pageInfo": Object {
                "endCursor": "b487224762b030f470967f45d7205823",
                "startCursor": "b487224762b030f470967f45d7205823",
              },
            },
            "coverImage": Object {
              "key": "image",
              "name": "Image",
              "sources": Array [
                Object {
                  "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=0241ED2F-B527-424C-917C-1142A398711F",
                },
              ],
            },
            "htmlContent": "<p>
	Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sollicitudin condimentum aliquet. In est nulla, lacinia ac dictum et, laoreet vitae elit. Proin tempus tellus ligula, a consequat diam consectetur a. Phasellus luctus velit sed lorem mollis commodo. Nunc sit amet blandit velit. Donec tincidunt congue facilisis. Sed iaculis at neque non porttitor. Phasellus ultrices egestas erat feugiat pellentesque. Duis venenatis, dolor quis fringilla tempus, sem lorem euismod lectus, sed egestas felis magna at felis. Pellentesque ut rhoncus erat, a pulvinar purus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Ut sit amet consequat est. Maecenas et porta dui, non condimentum lectus.</p>
<p>
	Suspendisse vel nibh odio. Pellentesque porta sapien ligula, in laoreet diam tempus sed. Morbi nunc erat, mattis eu pulvinar blandit, adipiscing quis magna. Ut quis dui lobortis velit suscipit consectetur. Nulla iaculis fermentum egestas. Aenean venenatis sagittis mauris, sed rhoncus purus accumsan ac. Suspendisse potenti. Sed sed tempor turpis. Duis sit amet nisi nec purus fringilla condimentum. Phasellus non lacus arcu. Donec scelerisque, erat sed tempor elementum, nulla risus scelerisque ante, ac imperdiet velit magna ut quam. Nam tristique orci auctor consequat laoreet. Quisque malesuada metus sed sodales eleifend. Aenean rhoncus, mi sit amet ullamcorper tincidunt, sem sem rutrum felis, in semper enim massa ut sem.</p>
<p>
	Vivamus diam urna, cursus in sapien in, porta gravida enim. Cras non fringilla arcu, tincidunt laoreet lacus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Aliquam volutpat felis quis augue faucibus ultrices. Morbi lobortis vestibulum sodales. Sed tincidunt urna vitae felis ultrices, pharetra placerat quam dignissim. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vel adipiscing tellus. In vitae sodales diam. Aliquam pharetra orci a porta molestie. In et neque bibendum, viverra leo sit amet, auctor magna. Morbi posuere massa sed metus euismod, et adipiscing sem dictum. Cras eget elementum risus, non imperdiet ligula.</p>
",
            "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
            "images": Array [
              Object {
                "__typename": "ImageMedia",
                "key": "image",
                "name": "Image",
                "sources": Array [
                  Object {
                    "__typename": "ImageMediaSource",
                    "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=0241ED2F-B527-424C-917C-1142A398711F",
                  },
                ],
              },
              Object {
                "__typename": "ImageMedia",
                "key": "detailImage",
                "name": "Detail Image",
                "sources": Array [
                  Object {
                    "__typename": "ImageMediaSource",
                    "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=3DA90982-118A-4BFE-9A32-58D9F610090A",
                  },
                ],
              },
              Object {
                "__typename": "ImageMedia",
                "key": "imageByUrlTest",
                "name": "Image By Url Test",
                "sources": Array [
                  Object {
                    "__typename": "ImageMediaSource",
                    "uri": "https://some-domain.com/some/path/to/image.jpg",
                  },
                ],
              },
              Object {
                "__typename": "ImageMedia",
                "key": "imageBySchemalessUrlTest",
                "name": "Image By Schemaless Url Test",
                "sources": Array [
                  Object {
                    "__typename": "ImageMediaSource",
                    "uri": "https://some-domain.com/some/path/to/image.jpg",
                  },
                ],
              },
              Object {
                "__typename": "ImageMedia",
                "key": "imageUnknownFormat",
                "name": "Image Of Unknown format",
                "sources": Array [
                  Object {
                    "__typename": "ImageMediaSource",
                    "uri": "ftp://some-domain.com/some/path/to/image.jpg",
                  },
                ],
              },
            ],
            "parentChannel": Object {
              "__typename": "ContentChannel",
              "id": "ContentChannel:744e1011cfa6a90bb2fbfeb3ecdc1451",
            },
            "sharing": Object {
              "__typename": "SharableContentItem",
              "message": "",
              "title": "SAMPLE: Easter",
              "url": "https://apollosrock.newspring.cc/",
            },
            "terms": Array [
              Object {
                "key": "summaryText",
                "value": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sollicitudin condimentum aliquet. In est nulla, lacinia ac dictum et, laoreet vitae elit. Proin tempus tellus ligula, a consequat diam consectetur a.",
              },
              Object {
                "key": "campuses",
                "value": "",
              },
              Object {
                "key": "primaryAudience",
                "value": "B364CDEE-F000-4965-AE67-0C80DDA365DC",
              },
              Object {
                "key": "secondaryAudiences",
                "value": "",
              },
              Object {
                "key": "speaker",
                "value": "Pete Foster",
              },
            ],
            "title": "SAMPLE: Easter",
            "videos": Array [
              Object {
                "__typename": "VideoMedia",
                "embedHtml": "<script src=\\"//fast.wistia.com/embed/medias/kqykx8xbzq.jsonp\\" async></script><script src=\\"//fast.wistia.com/assets/external/E-v1.js\\" async></script><div class=\\"wistia_responsive_padding\\" style=\\"padding:56.25% 0 0 0;position:relative;\\"><div class=\\"wistia_responsive_wrapper\\" style=\\"height:100%;left:0;position:absolute;top:0;width:100%;\\"><div class=\\"wistia_embed wistia_async_kqykx8xbzq videoFoam=true\\" style=\\"height:100%;width:100%\\">&nbsp;</div></div></div>",
                "key": "videoLink",
                "name": "Video Link",
                "sources": Array [
                  Object {
                    "__typename": "VideoMediaSource",
                    "uri": "https://rockrms.blob.core.windows.net/sampledata/podcasting/money-wise.mp4",
                  },
                ],
              },
            ],
          },
        },
      ],
    },
  },
}
`;

exports[`UniversalContentItem properly handles empty attribute values 1`] = `
Object {
  "data": Object {
    "node": Object {
      "__typename": "UniversalContentItem",
      "audios": Array [],
      "childContentItemsConnection": Object {
        "edges": Array [
          Object {
            "cursor": "b487224762b030f470967f45d7205823",
            "node": Object {
              "__typename": "UniversalContentItem",
              "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
            },
          },
        ],
        "pageInfo": Object {
          "endCursor": "b487224762b030f470967f45d7205823",
          "startCursor": "b487224762b030f470967f45d7205823",
        },
      },
      "coverImage": Object {
        "key": "image",
        "name": "Image",
        "sources": Array [
          Object {
            "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=0241ED2F-B527-424C-917C-1142A398711F",
          },
        ],
      },
      "htmlContent": "<p>
	Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sollicitudin condimentum aliquet. In est nulla, lacinia ac dictum et, laoreet vitae elit. Proin tempus tellus ligula, a consequat diam consectetur a. Phasellus luctus velit sed lorem mollis commodo. Nunc sit amet blandit velit. Donec tincidunt congue facilisis. Sed iaculis at neque non porttitor. Phasellus ultrices egestas erat feugiat pellentesque. Duis venenatis, dolor quis fringilla tempus, sem lorem euismod lectus, sed egestas felis magna at felis. Pellentesque ut rhoncus erat, a pulvinar purus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Ut sit amet consequat est. Maecenas et porta dui, non condimentum lectus.</p>
<p>
	Suspendisse vel nibh odio. Pellentesque porta sapien ligula, in laoreet diam tempus sed. Morbi nunc erat, mattis eu pulvinar blandit, adipiscing quis magna. Ut quis dui lobortis velit suscipit consectetur. Nulla iaculis fermentum egestas. Aenean venenatis sagittis mauris, sed rhoncus purus accumsan ac. Suspendisse potenti. Sed sed tempor turpis. Duis sit amet nisi nec purus fringilla condimentum. Phasellus non lacus arcu. Donec scelerisque, erat sed tempor elementum, nulla risus scelerisque ante, ac imperdiet velit magna ut quam. Nam tristique orci auctor consequat laoreet. Quisque malesuada metus sed sodales eleifend. Aenean rhoncus, mi sit amet ullamcorper tincidunt, sem sem rutrum felis, in semper enim massa ut sem.</p>
<p>
	Vivamus diam urna, cursus in sapien in, porta gravida enim. Cras non fringilla arcu, tincidunt laoreet lacus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Aliquam volutpat felis quis augue faucibus ultrices. Morbi lobortis vestibulum sodales. Sed tincidunt urna vitae felis ultrices, pharetra placerat quam dignissim. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vel adipiscing tellus. In vitae sodales diam. Aliquam pharetra orci a porta molestie. In et neque bibendum, viverra leo sit amet, auctor magna. Morbi posuere massa sed metus euismod, et adipiscing sem dictum. Cras eget elementum risus, non imperdiet ligula.</p>
",
      "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
      "images": Array [],
      "parentChannel": Object {
        "__typename": "ContentChannel",
        "id": "ContentChannel:744e1011cfa6a90bb2fbfeb3ecdc1451",
      },
      "sharing": Object {
        "__typename": "SharableContentItem",
        "message": "",
        "title": "SAMPLE: Easter",
        "url": "https://apollosrock.newspring.cc/",
      },
      "terms": Array [],
      "title": "SAMPLE: Easter",
      "videos": Array [],
    },
  },
}
`;
